### ISSUE TITLE
Latest version (0.5.2) requires gcc >= 7 to compile

### ISSUE THREAD

**Issue by jeffesquivels** [Venemo/node-lmdb#121]
It seems the latest version of lmdb-node uses the compiler flag [`-Wimplicit-fallthrough=2`](https://github.com/Venemo/node-lmdb/blob/master/binding.gyp#L29) which was [introduced in gcc version 7](https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/), so the compilation (and thus, the installation) of this package with older versions of gcc fails with an error like this:

```
make: Entering directory `/home/app/node_modules/node-lmdb/build'
  CC(target) Release/obj.target/node-lmdb/dependencies/lmdb/libraries/liblmdb/mdb.o
cc: error: unrecognized command line option '-Wimplicit-fallthrough=2'
make: Leaving directory `/home/app/node_modules/node-lmdb/build'
make: *** [Release/obj.target/node-lmdb/dependencies/lmdb/libraries/liblmdb/mdb.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.15.12-x86_64-linode105
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/app/node_modules/node-lmdb
gyp ERR! node -v v8.11.1
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
```

**Comment by Venemo** [comment]
Ouch. Sorry.

**Comment by Venemo** [comment]
I published a new version which removed the `-Wimplicit-fallthrough`. On the longer term, it would be nice if `-Wimplicit-fallthrough` would be there for GCC>=7 and not there otherwise. I'm not enough of a GYP magician to do that, though.

**Comment by Venemo** [comment]
@jeffesquivels Can you confirm that the problem is fixed now?

**Comment by jeffesquivels** [comment]
@Venemo I can confirm the problem is indeed fixed in the latest version (0.5.4).

>On the longer term, it would be nice if -Wimplicit-fallthrough would be there for GCC>=7 and not there otherwise. I'm not enough of a GYP magician to do that, though.

I also have an idea of how to accomplish this, if I get it to work I'll send you the PR.

Thanks!

**Comment by Venemo** [comment]
> I also have an idea of how to accomplish this, if I get it to work I'll send you the PR.

@jeffesquivels Awesome, thank you! Looking forward to it!
